pytorch_lightning/utilities/auto_restart.py

```diff
@@ -219,17 +222,140 @@ def __next__(self) -> Dict[str, Any]:
         return {"data": data, AutoRestartBatchKeys.PL_SAMPLERS: state_dicts}
 
     @staticmethod
-    def convert_batch_into_state_dict(batch) -> Dict[str, Dict[int, Any]]:
+    def _sanetize_batch_from_sampler_state(data: Any, state_dict: List):
+        elem_type = type(data)
+
+        # Recursively apply to collection items
+        if isinstance(data, Mapping):
+            out = []
+            for k, v in data.items():
+                if k == AutoRestartBatchKeys.PL_SAMPLERS:
+                    iterable_dataset_state_dict = {}
+                    batch_worker_id = v.pop("id")
+                    worker_id = batch_worker_id[-1].item()
+                    for sampler_name, sampler_state_dict in v.items():
+                        iterable_dataset_state_dict[sampler_name] = {
+                            worker_id: {
+                                "current_iteration": sampler_state_dict[worker_id]["current_iteration"][-1].item()
+                            }
+                        }
+                    state_dict.append(iterable_dataset_state_dict)
+                    return data["data"]
+                out.append((k, CaptureIterableDataset._sanetize_batch_from_sampler_state(v, state_dict)))
+            return elem_type(OrderedDict(out))
```

Comment:
is there a specific reason for using `OrderedDict` vs. just dict?

Reply:
Following apply_to_collection implementation. I believe there were a reason why OrderedDict was selected over it.